app/models/user.rb

```diff
@@ -4,11 +4,10 @@ class User < ApplicationRecord
   @@obj_info = %w(id name email wca_id country_iso2 avatar_url avatar_thumb_url gender birthdate delegate_status)
 
   has_many :subscriptions, -> { order(:payed_at) }
+  has_many :charges
```

Comment:
Maybe instead of having both `charges` and `subscriptions` you could change the existing `Subscription` model to include the stripe charge id and amount?

Reply:
Yeah, maybe just merging them is cleaner.